Classes/Image Upload/ImageUploadTableViewController.swift

```diff
@@ -235,3 +232,53 @@ extension ImageUploadTableViewController: UITextFieldDelegate {
         return false
     }
 }
+
+// MARK: UITextViewDelegate (Placeholder)
+
+extension ImageUploadTableViewController: UITextViewDelegate {
+    
+    private static let placeholderText = NSLocalizedString("Optional", comment: "")
+    
+    private func setBodyPlaceholderVisible(_ visible: Bool) {
+        if visible {
+            bodyTextField.text = ImageUploadTableViewController.placeholderText
+            bodyTextField.textColor = Styles.Colors.Gray.light.color
+            
+            setCursorToStart(textView: bodyTextField)
+        } else {
+            bodyTextField.text = nil
+            bodyTextField.textColor = Styles.Colors.Gray.dark.color
+        }
+    }
+    
+    private func setCursorToStart(textView: UITextView) {
+        textView.selectedTextRange = textView.textRange(
+            from: textView.beginningOfDocument,
+            to: textView.beginningOfDocument
+        )
+    }
+    
+    func textView(_ textView: UITextView, shouldChangeTextIn range: NSRange, replacementText text: String) -> Bool {
+        let currentText = textView.text as NSString
+        let updatedText = currentText.replacingCharacters(in: range, with: text)
+        
+        // If updated text view will be empty, add the placeholder and set
+        // the cursor to the beginning of the text view.
+        // Otherwise prepare for the user's entry.
+        if updatedText.isEmpty {
+            setBodyPlaceholderVisible(true)
+            
+            return false
+        } else if textView.textColor == Styles.Colors.Gray.light.color && !text.isEmpty {
+            setBodyPlaceholderVisible(false)
+        }
+        
+        return true
+    }
+    
+    func textViewDidChangeSelection(_ textView: UITextView) {
+        if self.view.window != nil && textView.textColor == Styles.Colors.Gray.light.color {
+            setCursorToStart(textView: textView)
+        }
+    }
+}
```

Comment:
Hmm I think we should maybe just move the placeholder `UILabel` from `MessageView` into `MessageTextView` and reuse that here? Would be a lot simpler. I can foresee some bugs from accidentally sending the placeholder text if actually its part of the `UITextView`.

Reply:
Totally agree. I will make appropriate changes.